sdk/keyvault/azure-keyvault-keys/tests/test_crypto_client.py

```diff
@@ -491,6 +491,67 @@ def test_calls_service_for_operations_unsupported_locally():
     assert supports_nothing.wrap_key.call_count == 0
 
 
+def test_local_only_mode_no_service_calls():
+    """A local-only CryptographyClient shouldn't call the service if an operation can't be performed locally"""
+
+    mock_client = mock.Mock()
+    jwk = mock.Mock(spec=JsonWebKey)
+    client = CryptographyClient.from_jkw(key_id="https://localhost/fake/key/version", jwk=jwk)
+    client._client = mock_client
+
+    supports_nothing = mock.Mock(supports=mock.Mock(return_value=False))
+    with mock.patch(CryptographyClient.__module__ + ".get_local_cryptography_provider", lambda *_: supports_nothing):
+        client.decrypt(EncryptionAlgorithm.rsa_oaep, b"...")
+    assert mock_client.decrypt.call_count == 0
+    assert supports_nothing.decrypt.call_count == 1
+
+    client.encrypt(EncryptionAlgorithm.rsa_oaep, b"...")
+    assert mock_client.encrypt.call_count == 0
+    assert supports_nothing.encrypt.call_count == 1
+
+    client.sign(SignatureAlgorithm.rs256, b"...")
+    assert mock_client.sign.call_count == 0
+    assert supports_nothing.sign.call_count == 1
+
+    client.verify(SignatureAlgorithm.rs256, b"...", b"...")
+    assert mock_client.verify.call_count == 0
+    assert supports_nothing.verify.call_count == 1
+
+    client.unwrap_key(KeyWrapAlgorithm.rsa_oaep, b"...")
+    assert mock_client.unwrap_key.call_count == 0
+    assert supports_nothing.unwrap_key.call_count == 1
+
+    client.wrap_key(KeyWrapAlgorithm.rsa_oaep, b"...")
+    assert mock_client.wrap_key.call_count == 0
+    assert supports_nothing.wrap_key.call_count == 1
+
+
+def test_local_only_mode_raise():
+    """A local-only CryptographyClient should raise an exception if an operation can't be performed locally"""
+
+    jwk = {"kty":"RSA", "key_ops":["decrypt", "verify", "unwrapKey"], "n":b"10011", "e":b"10001"}
+    client = CryptographyClient.from_jkw(key_id="https://localhost/fake/key/version", jwk=jwk)
+
+    # Algorithm not supported locally
+    with pytest.raises(NotImplementedError):
```

Comment:
Do you want to test some aspect of the error message, like that it includes the algorithm?

Reply:
I had considered it and decided not to, but it doesn't seem too implementation-dependent to expect this kind of error to return the algorithm name. I'll add that check, thanks!